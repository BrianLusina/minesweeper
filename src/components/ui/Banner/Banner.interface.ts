export type BannerProps = {
  title: string;
  description: string;
  link: string;
};
