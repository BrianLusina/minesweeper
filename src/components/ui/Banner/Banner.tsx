import { FunctionComponent } from 'react';
import { BannerProps } from './Banner.interface';

const Banner: FunctionComponent<BannerProps> = ({ title, description, link }) => {
  return (
    <section id="one">
      <header className="major">
        <h2>{title}</h2>
      </header>
      <p>{description}</p>
      <ul className="actions">
        <li>
          <a href={link} className="button">
            Learn More
          </a>
        </li>
      </ul>
    </section>
  );
};

export default Banner;
