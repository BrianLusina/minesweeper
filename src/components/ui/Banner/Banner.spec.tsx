import { render, screen } from '@testing-library/react';
import Banner from './Banner';

describe('Banner', () => {
  it('should render', () => {
    const title = 'Banner Title';
    const description = 'Banner description';
    const bannerLink = 'https://bannerlink.com';

    const { container } = render(
      <Banner title={title} description={description} link={bannerLink} />,
    );

    expect(container).toMatchSnapshot();

    const titleElement = screen.getByText(title);
    const descriptionElement = screen.getByText(description);

    expect(descriptionElement).toBeInTheDocument();
    expect(titleElement).toBeInTheDocument();
  });
});
