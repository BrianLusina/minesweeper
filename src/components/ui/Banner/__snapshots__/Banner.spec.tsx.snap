// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner should render 1`] = `
<div>
  <section
    id="one"
  >
    <header
      class="major"
    >
      <h2>
        Banner Title
      </h2>
    </header>
    <p>
      Banner description
    </p>
    <ul
      class="actions"
    >
      <li>
        <button
          class="button "
          type="button"
        >
          Learn More
        </button>
      </li>
    </ul>
  </section>
</div>
`;
